### PR TITLE
re-subscribe new head

### PR DESCRIPTION
**Pull request by 0xe3b0c4**
### Description
fix subscribe head timeout, re-subscribe new head

### CODE REVIEW

File: cmd/faucet/faucet.go [hunk 1]
```diff
@@ -745,17 +761,30 @@ func (f *faucet) loop() {
...
 		}
 	}()
+
 	// Wait for various events and assing to the appropriate background threads
+	subscribeTimeoutTimer := time.NewTimer(subscribeTimeout)
+
 	for {
 		select {
 		case head := <-heads:
+			if !subscribeTimeoutTimer.Stop() {
```
Comment (BillZong) [inline cmd/faucet/faucet.go:771]: No need to update timer here, all goes to the `update` chan below.

**Review by BillZong — APPROVED** [review]
1

File: cmd/faucet/faucet.go (hunk 1 above)
Comment (0xe3b0c4) [reply cmd/faucet/faucet.go:771]: no, you need check `Stop()` result

Comment (RichardBrown123go) [inline cmd/faucet/faucet.go:771]: No need to reset timer twice, since we'll write data to `update` chan, which will reset timer in only one place.

Comment (0xe3b0c4) [reply cmd/faucet/faucet.go:771]: ok, remove it [53132d8](https://github.com/antimatter-dao/bas-template-bsc/pull/18/commits/53132d8e1d5b002bb15d89f062ffdf2857501692)

**Review by RichardBrown123go — APPROVED** [review]
A nice solution 💯

**Pull request merged**